### PR TITLE
LPS-115417 SF

### PR DESCRIPTION
**Pull request by carolmariaabb**
Related issue: https://issues.liferay.com/browse/LPS-115417

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+			ddmFormFieldValue.setName(name);
+
+			DDMFormField ddmFormField = ddmFormFieldValue.getDDMFormField();
+
+			Value value = null;
+
+			if (ddmFormField.isLocalizable()) {
+				value = new LocalizedValue();
+
+				JSONObject valueJSONObject = jsonObject.getJSONObject("value");
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java:220]: @marcosapmf, can we get a `null` value as described from line 232 to 236?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+				for (String key : valueJSONObject.keySet()) {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java:222]: @marcosapmf, this is consistent, but if we rename `key` to `languageId` can be more meaningful. Do you think we should change? What's your opinion?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+				ddmFormValues.getDDMFormFieldValues()) {
+
+			DDMFormFieldTypeReportProcessor ddmFormFieldTypeReportProcessor =
+				_ddmFormFieldTypeReportProcessorTracker.
+					getDDMFormFieldTypeReportProcessor(
+						ddmFormFieldValue.getDDMFormField(),
+						ddmFormFieldValue.getType());
+
+			if (ddmFormFieldTypeReportProcessor != null) {
+				String fieldName = ddmFormFieldValue.getName();
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java:262]: @marcosapmf, please rename to `ddmFormFieldValueName`.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java (hunk 2 above)
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_7_3/UpgradeDDMFormInstanceReport.java:222]: @natocesarrego, I agree with you! I am going to make this change in the next pull :)

**Comment by marcosapmf** [comment]
> Nice job, @marcosapmf! 😄 The SQL queries were not so complex as you thought at first, right? 😁
> 
> Please check my comments below. 😉

Yeah, @natocesarrego. It was easier than I thought  :D

**Comment by marcosapmf** [comment]
@natocesarrego, please see https://github.com/natocesarrego/liferay-portal/pull/674

**Pull request closed without merging**